engine/src/main/java/org/camunda/bpm/engine/impl/pvm/runtime/LegacyBehavior.java

```diff
@@ -635,4 +639,34 @@ public boolean isFulfilled(PvmExecutionImpl element) {
     return walker.getCurrentElement();
   }
 
+  /**
+   * See #CAM-10978
+   * Use case process instance with <code>asyncBefore</code> startEvent
+   * After unifying the history variable's creation<br>
+   * The following changed:<br>
+   *   * variables will receive the <code>processInstanceId</code> as <code>activityInstanceId</code> in such cases (previously was the startEvent id)<br>
+   *   * historic details have new <code>initial</code> property to track initial variables that process is started with<br>
+   * The jobs created prior <code>7.13</code> and not executed before do not have historic information of variables.
+   * This method takes care of that.
+   */
+  public static void createMissingHistoricVariables(PvmExecutionImpl execution) {
+    Collection<VariableInstanceEntity> variables = ((ExecutionEntity) execution).getVariablesInternal();
+
+    if (variables != null && variables.size() > 0) {
+      // trigger historic creation if the history is not presented already
+      for (VariableInstanceEntity variable : variables) {
+
+        // processDefinitionId is introduced in 7.13 and for such variable we need to create history
+        if (variable.getProcessDefinitionId() == null &&
+            getHistoryLevel().isHistoryEventProduced(HistoryEventTypes.VARIABLE_INSTANCE_CREATE, variable)) {
```

Comment:
We can remove this second part, because that is also checked in the listener's `#onCreate` method.